src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
> Handle the scenario in the plugins: accept any exception and only process http ones, throw the rest (how about no?)

No, or i do not maintain this xD, it will just be a nightmare to handle.

> Create a RejectedPromise, which only passes http exceptions to the callback, rethrows everything else
> When the RejectedPromise is created, make sure only HTTP exceptions are injected into the Promise (although others are possible too) and everything else is thrown immediately (this would require to modify adapter promise implementations too)

:+1: and we must do the same for the FulfilledPromise, as actually it's accept anything, where we should only allow `ResponseInterface`